ludwig/config_validation/checks.py

```diff
@@ -606,3 +607,20 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
 
     if config.quantization and (not config.adapter or config.adapter.type != "lora"):
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
+
+
+@register_config_check
+def check_llm_template_references(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that prompt.template includes a valid reference to a feature column."""
+    if config.model_type != MODEL_LLM:
+        return
+
+    column_names = [feature.column for feature in config.input_features]
+    template_refs = set(Formatter().parse(config.prompt.template))
+    intersection = column_names & template_refs
+
+    if len(list(intersection)) == 0:
+        raise ConfigValidationError(
+            "Prompt template must include a reference to a column. This can be as simple as a string with just the "
+            'column and nothing else, e.g.: "{column_name}".'
+        )
```

Comment:
One thing to consider is that we actually allow no `template` to be specified if the `task` parameter of `prompt` is specified. In this case, we actually have a default zero shot and few shot prompt template that we use (zero shot also gets used for fine-tuning), and this automatically inserts `{__task__}` and `{__sample__}` into the data (I think).

1. Would be good to confirm if this is indeed what happens
2. Does this check allow for that to happen? I might add a conditional block to account for the task + template, or just task case.

Reply:
will add